core/src/main/java/org/ehcache/core/spi/service/DiskResourceService.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.core.spi.service;
+
+import org.ehcache.CachePersistenceException;
+import org.ehcache.spi.persistence.PersistableResourceService;
+
+/**
+ * Service to provide file based persistence context to caches tied to disk based stores.
```

Comment:
I would rephrase this class level doc (in place of the two paragraphs you have):

> Resource service handling file level operations for disk tiers.